### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by fatimah-z**
Changed the project structure according to the best practices. Separated models and routes into different files.Also created the config file to add configuration settings. Added dockerfile and docker-compose files for creating app and db containers. Added db migrations for sql database.

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    # if user exists
+    if user:
+        return Response("User Already exists", status=409)
+
+    # if users does not exist create new
+    new_user = User(email=email, password=generate_password_hash(password=password))
+    db.session.add(new_user)  # adding record to db session
+    db.session.commit()  # committing changes to db session
+
+    return Response("Signup Success", status=201)
```
Comment (AshfaqWanclouds) [inline app/routes.py]: kindly set status code for successful signed user

File: flask-project.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from app import app, db
+from app.models import User
+
+
+@app.shell_context_processor
+@app.shell_context_processor
+def make_shell_context():
+    return {'db': db, 'User': User}
+
```
Comment (AshfaqWanclouds) [inline flask-project.py]: should we add Car model here or not?

File: migrations/versions/5bef28ec7c00_.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.create_table('user',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('username', sa.String(length=64), nullable=True),
+    sa.Column('email', sa.String(length=120), nullable=True),
+    sa.Column('password', sa.String(length=128), nullable=True),
+    sa.PrimaryKeyConstraint('id')
+    )
```
Comment (AshfaqWanclouds) [inline migrations/versions/5bef28ec7c00_.py:27]: why car table is not created here by our migrations?

**Comment by AshfaqWanclouds** [comment]
We don't push some files on github

1. .flaskenv
2. .idea/
3. app.db  (database should not push to github)
4. migrations/alembic.ini
5. migrations/env.py
6. migrations/script.py.mako

**Review by maimoonaiqbal94 — COMMENTED** [review]
Delete __pycache folder

File: app/__init__.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+app = Flask(__name__)
+app.config.from_object(Config)
+db = SQLAlchemy()
+db.init_app(app)
+migrate = Migrate(app, db)
+
+
+from app import routes, models
+
+
```
Comment (maimoonaiqbal94) [inline app/__init__.py]: please follow pep8 standards.

File: app/models.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    email = db.Column(db.String(120), index=True, unique=True)
+    password = db.Column(db.String(128))
+
+
+class Car(db.Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    objectId = db.Column(db.String(25), nullable=False)
+    createdAt = db.Column(db.String(25), nullable=False)
+    updatedAt = db.Column(db.String(25), nullable=False)
```
Comment (maimoonaiqbal94) [inline app/models.py:25]: use snake case

File: app/__init__.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+from flask_migrate import Migrate
+
+
+app = Flask(__name__)
+app.config.from_object(Config)
+db = SQLAlchemy(app)
+from app import routes, models
+db.init_app(app)
+migrate = Migrate(app, db)
+
```
Comment (maimoonaiqbal94) [inline app/__init__.py]: too many new lines at end of file.

File: app/models.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+from app import db
+
+
+class User(db.Model):
+
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(64), index=True, unique=True)
```
Comment (maimoonaiqbal94) [inline app/models.py:9]: please add init methods here, and make sure to define tablenames here.

File: app/routes.py [hunk 8]
```diff
@@ -0,0 +1,38 @@
+from flask import request, Response
+from app.models import User
+from app import app, db
+from werkzeug.security import generate_password_hash, check_password_hash
+
+@app.route('/signup', methods=['POST'])
+def signup():
+    data = request.get_json()  # Retrieving the json data
+    username = data["username"]
```
Comment (maimoonaiqbal94) [inline app/routes.py:13]: add appropriate doc strings here.

**Review by maimoonaiqbal94 — APPROVED** [review]

**Pull request merged**